### PR TITLE
splitting google and google beta into 2 separate docs

### PR DESCRIPTION
**Pull request by megan07**
We want to generate different documentation for GA vs Beta when moving documentation to the registry. This is an initial attempt.

We do not want to merge this until we're ready to set up the documentation redirect to the registry, so that our current documentation is not overwritten.

The majority of this is changing the `hcl` examples to `terraform`, which should be super minor.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

File: third_party/terraform/website/docs/r/google_kms_key_ring_iam.html.markdown [hunk 1]
```diff
@@ -45,7 +45,7 @@ resource "google_kms_key_ring_iam_policy" "key_ring" {
 
 With IAM Conditions ([beta](https://terraform.io/docs/providers/google/provider_versions.html)):
```
Comment (c2thorn) [inline third_party/terraform/website/docs/r/google_kms_key_ring_iam.html.markdown:46]: A few beta IAM condition blocks

File: third_party/terraform/website/docs/r/healthcare_dicom_store_iam.html.markdown [hunk 2]
```diff
@@ -24,7 +24,7 @@ Three different resources help you manage your IAM policy for Healthcare DICOM s
```
Comment (c2thorn) [inline third_party/terraform/website/docs/r/healthcare_dicom_store_iam.html.markdown:24]: Just pointing out we have this block at the top here
```
~> **Warning:** These resources are in beta, and should be used with the terraform-provider-google-beta provider.
See [Provider Versions](https://terraform.io/docs/providers/google/guides/provider_versions.html) for more details on beta resources.	See [Provider Versions](https://terraform.io/docs/providers/google/guides/provider_versions.html) for more details on beta resources.
```

File: provider/core.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@ module Provider
   class Core
     include Compile::Core
 
-    def initialize(config, api, version_name, start_time)
```
Comment (c2thorn) [inline provider/core.rb:28]: Maybe set some more `doc_only=false` in method headers so you don't have to specify the variable in places where it doesn't seem relevant? Also should stop the cloud build from failing

**Comment by megan07** [comment]
We decided not to go this direction and instead copy docs from beta to ga at the end of the compiler. Closing.

**Pull request closed without merging**